packages/zero-react/src/use-zero-connection-state.tsx

```diff
@@ -0,0 +1,18 @@
+import {useSyncExternalStore} from 'react';
+import type {ConnectionState} from '../../zero-client/src/client/connection-manager.ts';
+import {useZero} from './zero-provider.tsx';
+
+/**
+ * Hook to subscribe to the connection status of the Zero instance.
+ *
+ * @returns The connection status of the Zero instance.
+ * @see {@link ConnectionState} for more details on the connection state.
+ */
+export function useZeroConnectionState(): ConnectionState {
```

Comment:
Added react/solid hooks for the connection state and deprecated the useZeroOnline